Remotion/Data/DomainObjects/Linq/MappingResolver.cs

```diff
@@ -156,10 +156,10 @@ public Expression ResolveTypeCheck (Expression checkedExpression, Type desiredTy
           throw new UnmappedItemException(message);
         }
 
-        var classDefinition = GetClassDefinition(desiredType);
+        var typeDefinition = GetTypeDefinition(desiredType);
         var idExpression = Expression.MakeMemberAccess(checkedExpression, s_idProperty.PropertyInfo);
         var classIDExpression = Expression.MakeMemberAccess(idExpression, s_classIDProperty.PropertyInfo);
-        var allClassDefinitions = EnumerableUtility.Singleton(classDefinition).Concat(classDefinition.GetAllDerivedClasses().Select(cd => cd));
+        var allClassDefinitions = typeDefinition.GetTypeHierarchy().OfType<ClassDefinition>();
```

Comment:
When we get the interfaces, we will need to add a distinct. Or change this to do get DerivedClasses + ImplementingClasses separately.